Exercise_2/main.cpp

```diff
@@ -0,0 +1,179 @@
+
+vector<nodes> read_from_file(const string &file){
+    string text;
+    vector<nodes> graph = {};
+    nodes node = {};
+    ifstream open_file(file);
+    if (!open_file){
+        cout << "File do not exists"<< endl;
+        return graph;
+    }
+    while(open_file>>node.start_node>>node.end_node>>node.distance){
+        graph.push_back(node);
+    }
+    return graph;
+}
+
+table find_min_vector(const vector<int> &list_of_visited_nodes, vector<table> &main_table){
+    table min_value_node = {};
+    int min_value = INT_MAX;
+    for(int i = 0;i< main_table.size(); ++i){
+        if (find(list_of_visited_nodes.begin(), list_of_visited_nodes.end(), main_table[i].vertex) ==
+        list_of_visited_nodes.end() && main_table[i].distance < min_value){
+            min_value = main_table[i].distance;
+            min_value_node = main_table[i];
+        }
+    }
+    return min_value_node;
+}
+
+void set_values_to_table(table current_node,
+                         const vector<nodes> &graph,
+                         vector<table> &main_table,
+                         vector<int> visited){
+    for(int i = 0; i < graph.size(); ++i){
+        for(int j = 0; j < main_table.size(); ++j){
+            if(current_node.vertex == graph[i].start_node
+               && find(visited.begin(), visited.end(), graph[i].end_node) == visited.end()
+               && graph[i].end_node == main_table[j].vertex
+               && current_node.distance+graph[i].distance < main_table[j].distance){
+                main_table[j].distance = graph[i].distance + current_node.distance;
+                main_table[j].previos_vector = graph[i].start_node;
+            }
+        }
+    }
+}
+
+list<table> shortest_path(const table &first_element, vector<table> &result_table, int start_node, int end_node){
+    list<table> graph_path = {};
+    int i = 0;
+    while(end_node != start_node){
+        if(result_table[i].vertex == end_node){
+            graph_path.push_front(result_table[i]);
+            end_node = result_table[i].previos_vector;
+            i=0;
+        }else{
+            ++i;
+        }
+    }
+    graph_path.push_front(first_element);
+    return graph_path;
+}
+
+
+table fill_up_result_table(const vector<int> &unvisited_vector,
+                           vector<table> &result_table,
+                           vector<nodes> &graph
+                           ){
+    vector<int> visited_vector = {};
+    table current_vertex = {};
+    table first_element = {};
+    int size = unvisited_vector.size();
+    while (visited_vector.size() != size) {
+        current_vertex = find_min_vector(visited_vector, result_table);
+        if (visited_vector.empty()) {
+            first_element = current_vertex;
+        }
+        visited_vector.push_back(current_vertex.vertex);
+        set_values_to_table(current_vertex, graph, result_table, visited_vector);
+    }
+    return first_element;
+}
+
+int check_input(vector<int> set_of_values, bool start) {
```

Comment:
Avoid copy using a const reference: `const std::vector<int>&`.

Instead of accepting the flag `start`, consider taking the 'prompt' as a string. Then user could call this function like this: `check_input(values, "Start node");`

So the signature should look like this:
`int check_input(const std::vector<int>& set_of_values, const std::string& prompt)`